packages/backend-defaults/src/entrypoints/rootHttpRouter/http/MiddlewareFactory.ts

```diff
@@ -137,18 +159,26 @@ export class MiddlewareFactory {
    * @returns An Express request handler
    */
   logging(): RequestHandler {
-    const logger = this.#logger.child({
-      type: 'incomingRequest',
-    });
-    const customMorganFormat =
-      '[:date[clf]] ":method :url HTTP/:http-version" :status :res[content-length] ":referrer" ":user-agent"';
-    return morgan(customMorganFormat, {
-      stream: {
-        write(message: string) {
-          logger.info(message.trimEnd());
+    const logger = this.#logger;
+    let meta: Record<string, string | undefined> = {};
+    return morgan(
+      (tokens: TokenIndexer, req: IncomingMessage, res: ServerResponse) => {
+        meta = getLogMeta(tokens, req, res);
```

Comment:
i gotta ask - is this a common pattern? it's a bit unexpected to mutate external state and then reuse it inside the stream thing. Feels like we can't for sure know that the stream is called exactly immediately after the first function? Could it end up being the case that the stream sends the wrong, old, meta?

Extreme thought ... what if we returned `JSON.stringify({ message: ..., a, b, c })` from the first function and then deserialize it in the stream? ;) Or is that just too crazy

Reply:
🤔 , not sure if it is a common patter as I'm doing it for the first time, but agree that we don't have control of when the stream method will be called. I updated the code with your suggest 🙌🏻